pvrparse: use video enumeration from eDVBVideo class

athoik opened the pull request:
This will prevent unwanted fails with hardcoded values in the future.

athoik committed:
pvrparse: use video enumeration from eDVBVideo class

This will prevent unwanted fails with hardcoded values in the future.

betacentauri commented:
I only don’t like this “m_streamtype < eDVBVideo::MPEG2” very much. What is smaller than mpeg2? With the “unknown” comment it’s understandable. Maybe use “m_streamtype == -1”, but this also not perfect. 
Let’s see what the others think. 
Thanks for looking into it!

WanWizard commented:
This is above my "paygrade", I don't have the knowledge to review this.

eriksl commented:
Agree with betacentauri. A set does conceptually not have an order, so comparing numerically does not have a semantic value. Even more, the contents of the set should not even be treated as having numerical representations, that's just a shortcoming of C(++) ;-)

eriksl commented:
I see a structure like this:

**if(streamtype < ...::mpeg2) // invalid**
{
    ....
}

switch(streamtype)
{
    case(...):
    ...
    case(...):
}

Why not rewrite this into something like:

switch(streamtype)
{
    case(...):
    ....
    case(...):
    ...
    case(...):
    ...
    **default:**
}

You can even have gcc warn or err on switches missing case statements for certain enum values.

pieterg commented:
note: the 'if () //invalid' is not part of the switch, it is applied before the switch to fill in the codec id if it is is still unknown at that point. So I'd not move that to a default clause, to maintain readability.

> You can even have gcc warn or err on switches missing case statements for certain enum values.

Indeed, that would help to make sure each newly added codec in the enum gets added into the switch as well.
But only if you omit the default clause.
(so we might have to get rid of that, it doesn't do anything except logging an error at the moment)

eriksl commented:
Even better, +1.

athoik committed:
decoder: add UNKNOWN codec to eDVBVideo

It will be used to pvrparse to mark unknown codec value.

athoik committed:
pvrparse: check all values from video enumeration

This commit drops the default case and uses all available codecs as suggested in https://github.com/OpenPLi/enigma2/pull/1914#issuecomment-450332617.
Also it uses the UNKNOWN enum value as suggsted in https://github.com/OpenPLi/enigma2/pull/1914#issuecomment-449665925

So once new codecs introduce we are going to have a warning on logs.

It would be nice to silence / fix all remaining warnigns and enable -Werror to force all warnings as errors.
So we can catch erlier un-noticed issues.

athoik committed:
pvrparse: add missing colon from previous commit

betacentauri commented:
Looks ok for me now.

athoik committed:
pvrparse: better handle UNKNOWN codec in setPid

As mentioned in the comments, we should parse UNKNOWN coded as well.

WanWizard commented:
So this can be merged?

athoik commented:
Let's wait another ok from @eriksl or @pieterg.

Then squash and merge!

eriksl commented:
I agree, let's wait for @pieterg

pieterg commented:
Looks good to me

eriksl commented:
Nice, thx all!
